modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageStore.java

```diff
@@ -235,19 +235,20 @@ void deleteMessages(final String storageQueueName, List<AndesMessageMetadata> me
     /**
      * get expired messages from store
      *
-     * @param limit max num of messages to read
+     * @param lowerBoundMessageID starting message Id of deletion safe zone
+     * @param  queueName Queue name
      * @return AndesRemovableMetadata
      * @throws AndesException
      */
-    List<AndesMessageMetadata> getExpiredMessages(int limit) throws AndesException;
+    List<AndesMessageMetadata> getExpiredMessages(long lowerBoundMessageID, String queueName) throws AndesException;
```

Comment:
Do we need to get a list of 'AndesMessageMetadata'. Isn't getting the message IDs enough?